### PR TITLE
Added new param to CustomImageCrop for new image fit types. 

### PR DESCRIPTION
**Pull request by JolimJostar**
New calculations to use for fitting/filling image inside screen/cropArea.

### CODE REVIEW

**Comment by JolimJostar** [comment]
Not in PR yet, but will add fillCropSpace

**Comment by JolimJostar** [comment]
I have trouble fixing [Format](https://app.travis-ci.com/github/icapps/flutter-custom-image-crop/jobs/604850736) test. 
Can you explain how to check if file formating is correct?

**Comment by ikbendewilliam** [comment]
@JolimJostar Thank you for your PR! We use `dart format .` to format our packages, normally running this command will fix the formatting.

**Comment by JolimJostar** [comment]
> @JolimJostar Thank you for your PR! We use `dart format .` to format our packages, normally running this command will fix the formatting.

Thank you

**Comment by ikbendewilliam** [comment]
Thank you! We'll look at it this week 😁

**Comment by g0dzillaa** [comment]
@JolimJostar Excellent mate! Precisely what I was looking for ❤️

**Comment by ikbendewilliam** [comment]
Very nice work!
One issue I see, the `CustomImageFit.fillCropHeight` and `CustomImageFit.fillVisibleHeight` work differently:
`fillCropHeight` uses the **height** of the screen to determine the cropheight. This is different from the current behaviour and changes the crop
```dart
case CustomImageFit.fillCropHeight:
  final cropSize = screenHeight;
  final cropSizeToPaint = cropSize * cropPercentage;
```
`fillVisibleHeight` uses the **width** of the screen to determine the cropheight. 
```dart
case CustomImageFit.fillVisibleHeight:
  final cropSize = screenHeight;
  final cropSizeToPaint = screenWidth * cropPercentage;
```
I would actually expect both to use the smallest size (width or height) as we don't want to change the crop depending on the imageFit IMO. 
Or at least act in a similar way, was this intentional? What do you expect here?

File: lib/src/calculators/calculate_crop_params.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      late final double cropSizeToPaint;
+      late final double defaultScale;
+
+      if (screenHeight > screenWidth) {
+        cropSize = screenHeight;
+        cropSizeToPaint = screenWidth * cropPercentage;
+        defaultScale = screenHeight / imageHeight;
+      } else {
+        cropSize = screenWidth;
+        cropSizeToPaint = cropSize * cropPercentage;
```
Comment (ikbendewilliam) [inline lib/src/calculators/calculate_crop_params.dart:67]: This works different than the `screenHeight > screenWidth` calculation. Is this intentional?

**Review by ikbendewilliam — CHANGES_REQUESTED** [review]

**Comment by JolimJostar** [comment]
> Very nice work! One issue I see, the `CustomImageFit.fillCropHeight` and `CustomImageFit.fillVisibleHeight` work differently: `fillCropHeight` uses the **height** of the screen to determine the cropheight. This is different from the current behaviour and changes the crop
> 
> ```dart
> case CustomImageFit.fillCropHeight:
>   final cropSize = screenHeight;
>   final cropSizeToPaint = cropSize * cropPercentage;
> ```
> 
> `fillVisibleHeight` uses the **width** of the screen to determine the cropheight.
> 
> ```dart
> case CustomImageFit.fillVisibleHeight:
>   final cropSize = screenHeight;
>   final cropSizeToPaint = screenWidth * cropPercentage;
> ```
> 
> I would actually expect both to use the smallest size (width or height) as we don't want to change the crop depending on the imageFit IMO. Or at least act in a similar way, was this intentional? What do you expect here?

I will look into this. Sorry for delayed response.

File: lib/src/calculators/calculate_crop_params.dart (hunk 1 above)
Comment (JolimJostar) [reply lib/src/calculators/calculate_crop_params.dart]: same as last conversation

**Review by ikbendewilliam — APPROVED** [review]

**Pull request merged**